### PR TITLE
[CORE] Use spark.io.compression.codec if spark.gluten.sql.columnar.shuffle.c…

### PR DESCRIPTION
**Pull request by marin-ma**
…odec not set

### CODE REVIEW

File: gluten-core/src/main/scala/io/glutenproject/execution/ShuffleUtils.scala [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        s"The value of $codecConf should be one of " +
+          s"${validValues.mkString(", ")}, but was $codec")
+    }
+  }
+
+  def getCompressionCodec(conf: SparkConf): String = {
+    val glutenConfig = GlutenConfig.getConf
+    glutenConfig.columnarShuffleCodec match {
+      case Some(codec) =>
+        val glutenCodecConf = "spark.gluten.sql.columnar.shuffle.codec"
```
Comment (jinchengchenghh) [inline gluten-core/src/main/scala/io/glutenproject/execution/ShuffleUtils.scala:39]: Please use `GlutenConfig.COLUMNAR_SHUFFLE_CODEC.key`

File: gluten-core/src/main/scala/io/glutenproject/execution/ShuffleUtils.scala [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+          checkCodecValues(glutenCodecConf, codec, GlutenConfig.GLUTEN_QAT_SUPPORTED_CODEC)
+          GlutenConfig.GLUTEN_QAT_CODEC_PREFIX + codec
+        } else if (glutenConfig.columnarShuffleEnableIaa) {
+          checkCodecValues(glutenCodecConf, codec, GlutenConfig.GLUTEN_IAA_SUPPORTED_CODEC)
+          GlutenConfig.GLUTEN_IAA_CODEC_PREFIX + codec
+        } else {
+          codec
+        }
+      case None =>
+        val sparkCodecConf = "spark.io.compression.codec"
```
Comment (jinchengchenghh) [inline gluten-core/src/main/scala/io/glutenproject/execution/ShuffleUtils.scala:50]: so as this one

**Review by jinchengchenghh — APPROVED** [review]

**Comment by marin-ma** [comment]
Run Gluten Clickhouse CI

**Comment by marin-ma** [comment]
Run Gluten Clickhouse CI

**Comment by marin-ma** [comment]
Run Gluten Clickhouse CI

**Comment by zzcclp** [comment]
there are some ch ut using snappy as shuffle codec, can it add snappy into the supported codec list ?

**Pull request merged**